.github/workflows/shellcheck.yml

```diff
@@ -16,10 +16,6 @@ jobs:
       # required for all workflows
       security-events: write
 
-      # only required for workflows in private repositories
-      actions: read
-      contents: read
```

Comment:
Why are these unnecessary?

Reply:
I thought we don't use "private repositories"... (and removing them did not break the test...)

Reply:
I'm not sure what that means, but I recently added permissions based on a suggestion from the journald team to all of our github actions - https://github.com/linux-system-roles/.github/pull/8/files

Reply:
It seems the examples listed here do not have them... (but the permissions are not identical to each other...)
https://github.com/redhat-plumbers-in-action/differential-shellcheck#real-life-examples-of-usage

Reply:
I removed the entire permissions section from the `lint job`, but the test ran and failed as expected...
```
    permissions:
      # required for all workflows
      security-events: write

      # only required for workflows in private repositories
      actions: read
      contents: read
```

Reply:
I'm not exactly sure how permissions work - but I think the general idea is that the default permissions could allow too much - better to follow the model of the other github actions that use a `permission: contents: read` https://github.com/linux-system-roles/logging/pull/320/files#diff-eace62a8ebcdafd0f6975200d68cbf851300952aae47db07e71f383038a368b3R8 for the whole action - then relax that permission for individual jobs where needed (using `security-events: write` - perhaps we don't need the others)

Reply:
Here are the types of REST actions allowed by `security-events: read|write` https://docs.github.com/en/rest/overview/permissions-required-for-github-apps?apiVersion=2022-11-28#security-events
It may be that the action cannot update `Code scanning` if it doesn't have `security-events: write` permission